docs/docs/adding-search-with-algolia.md

```diff
@@ -2,593 +2,612 @@
+
+```js:title=gatsby-config.js
+require("dotenv").config()
+```
 
-```text:title=.env.example
-# rename this file to .env and supply the values listed below
-# also make sure they are available to the build tool (e.g. Netlify)
-# warning: variables prefixed with GATSBY_ will be made available to client-side code
-# be careful not to expose sensitive data (in this case your Algolia admin key)
+Then add the configuration for `gatsby-plugin-algolia` to the list of `plugin`s in the `gatsby-config.js`. `dotenv` has made the configuration values available as keys in `process.env`.
 
-GATSBY_ALGOLIA_APP_ID=insertValue
-GATSBY_ALGOLIA_SEARCH_KEY=insertValue
-ALGOLIA_ADMIN_KEY=insertValue
+```js:title=gatsby-config.js
+  plugins: [
+    ... // your existing plugins here
+    {
+      resolve: `gatsby-plugin-algolia`,
+      options: {
+        appId: process.env.GATSBY_ALGOLIA_APP_ID,
+        apiKey: process.env.ALGOLIA_ADMIN_KEY,
+        queries: require("./src/utils/algolia-queries")
+      },
+    }
+  ],
+}
 ```
 
-The `queries` allow you to grab the data you want Algolia to index directly from Gatsby's GraphQL layer by exporting from `src/utils/algolia.js` an array of objects, each containing a required GraphQL query and an optional index name, transformer function and settings object.
+### Query the pages for indexing
 
-> Note: The excerpts below are pruned to 5000 characters. This is because, by default, Algolia has an upper bound of 10KB for an index entry. If you hit this boundary, you'll be presented with an error like this: `AlgoliaSearchError: Record at the position XX objectID=xx-xx-xx-xx-xx is too big size=10356 bytes. Contact us if you need an extended quota`. Make sure that your overall record is within this limit.
+You still need to supply `queries` configuration. Queries tell the Algolia plugin what data is to be indexed. They perform GraphQL queries for the relevant pages and convert the response into a set of Algolia records. These contain key/value pairs with the data to be indexed.
+
+This could have been entered straight into the `gatsby-config.js`, but to avoid clutter the configuration above loads it from the file `src/utils/algolia-queries.js`.
+
+```js:title=src/utils/algolia-queries.js
+const pagePath = "src/markdown-pages"
+const indexName = "Pages"
 
-```js:title=src/utils/algolia.js
 const pageQuery = `{
   pages: allMarkdownRemark(
     filter: {
-      fileAbsolutePath: { regex: "/pages/" },
-      frontmatter: {purpose: {eq: "page"}}
-    }
-  ) {
-    edges {
-      node {
-        objectID: id
-        frontmatter {
-          title
-          slug
-        }
-        excerpt(pruneLength: 5000)
-      }
+      fileAbsolutePath: { regex: "/${pagePath.replace("/", "/")}/" },
     }
-  }
-}`
-
-const postQuery = `{
-  posts: allMarkdownRemark(
-    filter: { fileAbsolutePath: { regex: "/posts/" } }
   ) {
     edges {
       node {
-        objectID: id
+        id
```

Comment:
The text below says "each record must have an ID in the key `objectID`", and it throws an error without it. So this needs to stay.
```suggestion
        objectID: id
```